src/Services/JsonApi.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Dxw\Whippet\Services;
+
+// Responsible for:
+// - calling out to a basic API
+// - parsing JSON
+// - converting error codes
+class JsonApi
+{
+    public function __construct(
+        \Dxw\Whippet\Services\BaseApi $base_api
+    ) {
+        $this->baseApi = $base_api;
+    }
+
+    public function get($url)
+    {
+        $response = $this->baseApi->get($url);
+
+        if ($response->isErr()) {
+            return $response;
+        }
+
+        $response_as_array = json_decode($response->unwrap(), true);
+        if (is_null($response_as_array)) {
+            return \Result\Result::err('Received invalid JSON when requesting '.$url);
```

Comment:
I've clarified this error message to say that we couldn't parse the JSON, and also added some code to check the JSON response, so we'll get errors if the JSON is not as expected.